src/debonair/Restaurant.java

```diff
@@ -0,0 +1,137 @@
+// user defined package
+package debonair;
+
+import java.time.LocalDateTime;
+// imports from in-built packages
+import java.util.Iterator;
+import java.util.Scanner;
+
+// importing classes from other packages
+import resources.Constants;
+
+// Restaurant class
+public class Restaurant implements IRestaurant {
+	/**
+	 * This method is used to show menu to the customer
+	 */
+	public void showMenu() {
+		System.out.println("The menu has following items:-");
+		// Iterator created to print items from Menu class's itemMenu List
+		Iterator<Item> itemsIterator = Menu.ITEM_MENU.iterator();
+		while (itemsIterator.hasNext()) {
+			Item item = itemsIterator.next();
+			System.out.println(item.toString());
+		}
+	}
+
+	/**
+	 * This method will be used to take order from the customer
+	 */
+	public void takeOrder() {
+		Order.createOrder();
```

Comment:
Refer to comment in order class

Reply:
Ask for Customer information from user .also first check if he is a new or old customer and do proper handling of all scenarios.

Reply:
Apply discounts to the order based on the discounts applicable to customer